agera/src/main/java/com/google/android/agera/Functions.java

```diff
@@ -110,5 +133,47 @@ public T apply(@NonNull F from) {
     }
   }
 
+  private static final class ParallelMapFunction<F, T> implements Function<List<F>, List<T>> {
+    @NonNull
+    private final Function<F, T> function;
+    @NonNull
+    private final ExecutorService executorService;
+
+    ParallelMapFunction(@NonNull final ExecutorService executorService,
+        @NonNull final Function<F, T> function) {
+      this.executorService = checkNotNull(executorService);
+      this.function = checkNotNull(function);
+    }
+
+    @NonNull
+    @Override
+    @SuppressWarnings("unchecked")
+    public List<T> apply(@NonNull final List<F> input) {
+      if (input.isEmpty()) {
+        return emptyList();
+      }
+      final List<T> result = new ArrayList(input.size());
+      final List<Future<T>> futureResult = new ArrayList(input.size());
```

Comment:
Hm maybe. it's really just a convenience method, not sure if there's much value left when returning a future.
Trying to envision the usage here. I guess one transform of this method must then be followed by another that does future.get either way? no?